website/archiver/tasks.py

```diff
@@ -146,23 +139,45 @@ def archive_addon(addon_short_name, job_pk, stat_result):
     folder_name = src_provider.archive_folder_name
     provider = src_provider.config.short_name
     cookie = user.get_or_create_cookie()
-    data = {
-        'source': {
-            'cookie': cookie,
-            'nid': src._id,
-            'provider': provider,
-            'path': '/',
-        },
-        'destination': {
-            'cookie': cookie,
-            'nid': dst._id,
-            'provider': settings.ARCHIVE_PROVIDER,
-            'path': '/',
-        },
-        'rename': folder_name,
-    }
     copy_url = settings.WATERBUTLER_URL + '/ops/copy'
-    make_copy_request.delay(job_pk=job_pk, url=copy_url, data=data)
+    if addon_short_name == 'dataverse':
```

Comment:
There's a fair amount of duplication below that can be removed. Also, any special-casing that needs to be done (e.g. with an addon) should be accompanied by a comment explaining why it's necessary.